static/src/javascripts/projects/common/modules/commercial/membership-engagement-banner.js

```diff
@@ -0,0 +1,214 @@
+        defaultParams,
+        getUserVariantParams(userVariant, campaignId, defaultParams.offering)
+    );
+};
+
+// Used to send an interaction if the engagement banner is shown.
+// #? A lot of null checking here, but it's hard to reason about the possible values
+//    `interaction` could take. Perhaps we can simplify when we have more Flow coverage
+const recordInteraction = (interaction: ?Interaction): void => {
+    if (interaction) {
+        const { component, value } = interaction;
+
+        if (component && value) {
+            ophan.record({
+                component,
+                value,
+            });
+        }
+    }
+};
+
+const selectSequentiallyFrom = (array: Array<string>): string =>
+    array[local.get('gu.alreadyVisited') % array.length];
+
+const showBanner = (params: Object): void => {
+    if (isBlocked()) {
+        return;
+    }
+
+    const paypalAndCreditCardImage =
+        config.get('images.acquisitions.paypal-and-credit-card') || '';
+    const colourClass = params.colourStrategy();
+    const messageText = Array.isArray(params.messageText)
+        ? selectSequentiallyFrom(params.messageText)
+        : params.messageText;
+    const urlParameters = {
+        REFPVID: params.pageviewId,
+        INTCMP: params.campaignCode,
+    };
+    const linkUrl = `${params.linkUrl}?${constructQuery(urlParameters)}`;
+    const buttonCaption = params.buttonCaption;
+    const buttonSvg = inlineSvg('arrowWhiteRight');
+    const renderedBanner = `
+    <div id="site-message__message">
+        <div class="site-message__message site-message__message--membership">
+            <span class = "membership__message-text">${messageText}</span>
+            <span class="membership__paypal-container">
+                <img class="membership__paypal-logo" src="${paypalAndCreditCardImage}" alt="Paypal and credit card">
+                <span class="membership__support-button"><a class="message-button-rounded__cta ${colourClass}" href="${linkUrl}">${buttonCaption}${buttonSvg}</a></span>
+            </span>
+        </div>
+        <a class="u-faux-block-link__overlay js-engagement-message-link" target="_blank" href="${linkUrl}" data-link-name="Read more link"></a>
+    </div>`;
+
+    const messageShown = new Message(messageCode, {
+        pinOnHide: false,
+        siteMessageLinkName: 'membership message',
+        siteMessageCloseBtn: 'hide',
+        siteMessageComponentName: params.campaignCode,
+        trackDisplay: true,
+        cssModifierClass: colourClass,
+    }).show(renderedBanner);
+
+    if (messageShown) {
+        recordInteraction(params.interactionOnMessageShow);
+
+        mediator.emit('membership-message:display');
+    }
+};
+
+const membershipEngagementBannerInit = (): Promise<void> =>
+    get().then(location => {
+        const bannerParams = deriveBannerParams(location);
+
+        if (
+            bannerParams &&
+            (local.get('gu.alreadyVisited') || 0) >= bannerParams.minArticles
```

Comment:
Since `'gu.alreadyVisited'` is used in two places, would it be better to store it in a variable at the top? Or have a wrapper, which returns the localStorage entry?